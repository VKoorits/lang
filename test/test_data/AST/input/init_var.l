sub factorial(n):
	if n < 2 :
		retunr 1
	return n * factorial(n-1)

var str = "hello" + "world", num = factorial(5)
var lst = [1,2,3,{"k" => "v"}, 8]
