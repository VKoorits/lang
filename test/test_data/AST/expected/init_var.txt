INIT_0 INIT_1 NULL
INIT_0 :str ELEM_2 num ELEM_3 NULL
ELEM_2 :hello world + NULL
ELEM_3 :CALL_4 NULL
CALL_4 :LIST_5 factorial NULL
LIST_5 :ELEM_6 NULL
ELEM_6 :5 NULL
INIT_1 :lst ELEM_2 NULL
ELEM_2 :ARR_3 NULL
ARR_3 :ELEM_4 ELEM_5 ELEM_6 ELEM_7 ELEM_8 NULL
ELEM_4 :1 NULL
ELEM_5 :2 NULL
ELEM_6 :3 NULL
ELEM_7 :HASH_9 NULL
HASH_9 :ELEM_10 ELEM_11 NULL
ELEM_10 :k NULL
ELEM_11 :v NULL
ELEM_8 :8 NULL
FUNCTION: 
	head: 
factorial ( n ) NULL
	body: 
EXPR_0 T_BODY_1 if return n CALL_2 * NULL
EXPR_0 :n 2 < NULL
T_BODY_1 :retunr 1 NULL
CALL_2 :LIST_3 factorial NULL
LIST_3 :ELEM_4 NULL
ELEM_4 :n 1 - NULL
END
